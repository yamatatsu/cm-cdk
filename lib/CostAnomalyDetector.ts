import { App, Stack, StackProps, aws_sns, aws_ce } from "aws-cdk-lib";

export class CostAnomalyDetector extends Stack {
  public readonly topic: aws_sns.ITopic;

  constructor(scope: App, id: string, props?: StackProps) {
    super(scope, id, props);

    const topic = new aws_sns.Topic(this, "topic");

    // デプロイできない。。。
    // Error [ValidationError]: Template format error: Unrecognized resource types: [AWS::CE::AnomalyMonitor]
    // const anomalyMonitor = new aws_ce.CfnAnomalyMonitor(
    //   this,
    //   "AnomalyMonitor",
    //   {
    //     monitorName: "MyAnomalyMonitor",
    //     monitorType: "DIMENSIONAL",
    //     monitorDimension: "SERVICE",
    //   }
    // );

    // const anomalySubscription = new aws_ce.CfnAnomalySubscription(
    //   this,
    //   "AnomalySubscription",
    //   {
    //     subscriptionName: "MyAnomalySubscription",
    //     frequency: "IMMEDIATE",
    //     monitorArnList: [anomalyMonitor.ref],
    //     subscribers: [{ type: "SNS", address: topic.topicArn }],
    //     // 実験的に1を設定してみる
    //     threshold: 1,
    //   }
    // );

    // anomalySubscription.addDependsOn(anomalyMonitor);

    this.topic = topic;
  }
}
