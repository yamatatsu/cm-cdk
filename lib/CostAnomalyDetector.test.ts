import { SynthUtils } from "@aws-cdk/assert";
import { App } from "aws-cdk-lib";
import { CostAnomalyDetector } from "./CostAnomalyDetector";

test("snapshot test", () => {
  const app = new App();

  const target = new CostAnomalyDetector(app, "Target", {});

  expect(SynthUtils.toCloudFormation(target)).toMatchSnapshot();
});
