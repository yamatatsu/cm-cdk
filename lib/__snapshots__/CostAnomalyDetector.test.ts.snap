// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
Object {
  "Resources": Object {
    "topic69831491": Object {
      "Type": "AWS::SNS::Topic",
    },
  },
}
`;
